From 4bac50c2c94023cb1b5bf947abfb1c72eeeb12d5 Mon Sep 17 00:00:00 2001
From: Cupertino Miranda <cmiranda@synopsys.com>
Date: Fri, 2 Mar 2018 17:33:48 +0100
Subject: [PATCH] [ARC] Improved robustness. Return FALSE in case of NULL
 pointer.

bfd/
2018-03-01  Cupertino Miranda <cmiranda@synopsys.com>

	* elf32-arc.c (elf_arc_finish_dynamic_symbol) Return FALSE in case
	arc_htab is NULL.

Signed-off-by: Alexey Brodkin <abrodkin@synopsys.com>
---
 bfd/elf32-arc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/bfd/elf32-arc.c b/bfd/elf32-arc.c
index 166a5ef604ff..72b808b99127 100644
--- a/bfd/elf32-arc.c
+++ b/bfd/elf32-arc.c
@@ -2403,6 +2403,9 @@ elf_arc_finish_dynamic_symbol (bfd * output_bfd,
     {
       struct elf_arc_link_hash_table *arc_htab = elf_arc_hash_table (info);
 
+      if(arc_htab == NULL)
+	return FALSE;
+
       if (h->dynindx == -1
 	  || (h->root.type != bfd_link_hash_defined
 	      && h->root.type != bfd_link_hash_defweak)
-- 
2.17.1

